### PR TITLE
More microfacet distribution functions work

### PR DESCRIPTION
**Pull request by est77**
- Small refactors and other changes in new MDF classes.
- Fixed some bugs in new sampling method.
- Removed the old microfacet classes.
- Updated reference images.

### CODE REVIEW

File: src/appleseed/CMakeLists.txt [hunk 1]
```diff
@@ -346,7 +345,7 @@ set (foundation_meta_benchmarks_sources
     foundation/meta/benchmarks/benchmark_knn.cpp
     foundation/meta/benchmarks/benchmark_math_filter.cpp
     foundation/meta/benchmarks/benchmark_matrix.cpp
-    foundation/meta/benchmarks/benchmark_microfacet.cpp
+    #foundation/meta/benchmarks/benchmark_microfacet.cpp
```
Comment (dictoon) [inline src/appleseed/CMakeLists.txt]: What's up with that?

File: src/appleseed/foundation/math/microfacet.h [hunk 2]
```diff
@@ -27,14 +26,22 @@
 // THE SOFTWARE.
 //
 
-#ifndef APPLESEED_FOUNDATION_MATH_MICROFACET_H
-#define APPLESEED_FOUNDATION_MATH_MICROFACET_H
+#ifndef APPLESEED_FOUNDATION_MATH_MICROFACET2_H
```
Comment (dictoon) [inline src/appleseed/foundation/math/microfacet.h]: `APPLESEED_FOUNDATION_MATH_MICROFACET_H`

File: src/appleseed/foundation/math/microfacet.h [hunk 3]
```diff
@@ -81,284 +477,553 @@ class BlinnMDF
...
+        }
+
+        Vector<T, 2> slope;
+
+        // Precomputations.
+        const T tan_theta = sin_theta / cos_theta;
+        const T tan_theta2 = square(tan_theta);
+        const T cot_theta = T(1.0) / tan_theta;
+
+        //const T lambda = (T(-1.0) + std::sqrt(T(1.0) + tan_theta2)) * T(0.5);
```
Comment (dictoon) [inline src/appleseed/foundation/math/microfacet.h]: Remove dead code?

File: src/appleseed/renderer/modeling/bsdf/kelemenbrdf.cpp [hunk 4]
```diff
@@ -72,6 +72,47 @@ namespace renderer
 
 namespace
 {
+
+    template <typename T>
+    class WardMDFAdapter
+    {
+      public:
+        explicit WardMDFAdapter(const T alpha)
+            : m_alpha(alpha)
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/kelemenbrdf.cpp]: Indent by two spaces.

File: src/appleseed/renderer/modeling/bsdf/kelemenbrdf.cpp [hunk 5]
```diff
@@ -607,8 +648,10 @@ namespace
             const double        m,
             const Spectrum&     rs)
         {
-            generate_specular_albedo_plot_data("Ward", m, WardMDF<double>(m), rs);
-            generate_specular_albedo_plot_data("Beckmann", m, BeckmannMDF<double>(m), rs);
+            generate_specular_albedo_plot_data("Ward", m, WardMDFAdapter<double>(m), rs);
+
+            // TODO: not really sure what to do with this... (est.)
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/kelemenbrdf.cpp]: What's the problem?

Comment (est77) [inline src/appleseed/renderer/modeling/bsdf/kelemenbrdf.cpp]: Beckmann cannot be sampled anymore using the old methods.
I'd need to recover the old implementation from the repository and copy the code here.
It is worth it?

File: src/appleseed/CMakeLists.txt (hunk 1 above)
Comment (est77) [inline src/appleseed/CMakeLists.txt]: It will come back in a future PR.

File: src/appleseed/renderer/modeling/bsdf/kelemenbrdf.cpp (hunk 5 above)
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/kelemenbrdf.cpp]: Can Beckmann be used at all with the Kelemen BRDF?

Comment (est77) [inline src/appleseed/renderer/modeling/bsdf/kelemenbrdf.cpp]: Apparently it's only used to generate some graphs for testing.

**Pull request merged**